app/controllers/api/v1/api_controller.rb

```diff
@@ -0,0 +1,20 @@
+class Api::V1::ApiController < ActionController::API
+  respond_to :json
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found_error
+  rescue_from ActionController::ParameterMissing, with: :parameter_missing_error
+
+  protected
+
+  def check_company_token
+    params.require(:company_token)
+    @company = Company.find_by!(token: params[:company_token])
+  end
+
+  def not_found_error
+    head 404
+  end
```

Comment:
Se quiserem vocês podem customizar a mensagem de erro com algo tipo:

```suggestion
  def not_found_error(exception)
    render status: :not_found, 
           json: "#{controller_name
                      .classify
                      .constantize
                      .model_name
                      .human} não encontrado"
  end
```

Reply:
Boa sugestão! Implementamos.